src/Integration.UnitTests/LocalServices/ErrorListInfoBarControllerTests.cs

```diff
@@ -0,0 +1,815 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ErrorListInfoBarControllerTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.InfoBar;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using System;
+using System.ComponentModel.Composition.Primitives;
+using System.Linq;
+using System.Windows.Threading;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
```

Comment:
There's twice as much of code here as in the test subject just due to all the edge cases, partial list:
1) clicking Update twice
2) The user disconnected after the info bar have been shown
3) The user connected to a different server after the info bar have been shown
4) The user bound to a different project after the info bar have been shown
5) Build/Debug started (the underlying commands will return false when asked if can execute)